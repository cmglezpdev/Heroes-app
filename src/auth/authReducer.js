import { types } from "../types/types";

// const state = {
//     name: 'Carlos',
//     logged: true
// }


export const authReducer = ( state = {}, action ) => {

    switch( action.type ) {
        case types.login:
            return {
                ...action.payload,
                logged: true
            }

        case types.logout:
            return {
                logged: false
            }

        default:
            return state;
    }
}