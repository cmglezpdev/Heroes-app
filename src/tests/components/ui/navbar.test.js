/* eslint-disable testing-library/no-render-in-setup */
import { fireEvent, render, screen } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom";
import { AuthContext } from "../../../auth/authContext"
import { Navbar } from "../../../components/ui/NavBar"
import { types } from "../../../types/types";

const mockDispatch = jest.fn();
const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate,
}));

describe('tests about navbar', () => { 
    
    const contextValue = {
        user: {
            name: 'Carlos Manuel',
            logged: true
        },
        dispatch: mockDispatch
    }

    let container; 

    beforeEach(() => {
        container = render(
            <AuthContext.Provider value={contextValue}>
                <MemoryRouter>
                    <Navbar />
                </MemoryRouter>
            </AuthContext.Provider>
        )
    })


    test('should show it correctly', () => {
        expect( container ).toMatchSnapshot();
        screen.getByText('Carlos Manuel');
        screen.getByText('Logout');
    });

    test('should call the navigate and the dispatch with the args in the function of logout', () => {
        const logoutBtn = screen.getByRole('button');
        fireEvent.click(logoutBtn);

        expect( mockDispatch ).toHaveBeenCalledTimes(1);
        expect( mockDispatch ).toHaveBeenCalledWith({ type: types.logout })
        expect( mockNavigate ).toHaveBeenCalledTimes(1);
        expect( mockNavigate ).toHaveBeenCalledWith('/login', {replace: true});
    })
 })