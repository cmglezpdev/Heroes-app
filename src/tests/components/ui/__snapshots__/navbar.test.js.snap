// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests about navbar should show it correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbar navbar-expand-sm navbar-dark bg-dark"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          Asociaciones
        </a>
        <div
          class="navbar-collapse"
        >
          <div
            class="navbar-nav"
          >
            <a
              class="nav-item nav-link "
              href="/marvel"
            >
              Marvel
            </a>
            <a
              class="nav-item nav-link "
              href="/dc"
            >
              DC
            </a>
            <a
              class="nav-item nav-link "
              href="/search"
            >
              Search
            </a>
          </div>
        </div>
        <div
          class="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
        >
          <ul
            class="navbar-nav ml-auto"
          >
            <span
              class="nav-link nav-item text-info"
            >
              Carlos Manuel
            </span>
            <button
              class="nav-item nav-link btn"
            >
              Logout
            </button>
          </ul>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbar navbar-expand-sm navbar-dark bg-dark"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        Asociaciones
      </a>
      <div
        class="navbar-collapse"
      >
        <div
          class="navbar-nav"
        >
          <a
            class="nav-item nav-link "
            href="/marvel"
          >
            Marvel
          </a>
          <a
            class="nav-item nav-link "
            href="/dc"
          >
            DC
          </a>
          <a
            class="nav-item nav-link "
            href="/search"
          >
            Search
          </a>
        </div>
      </div>
      <div
        class="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
      >
        <ul
          class="navbar-nav ml-auto"
        >
          <span
            class="nav-link nav-item text-info"
          >
            Carlos Manuel
          </span>
          <button
            class="nav-item nav-link btn"
          >
            Logout
          </button>
        </ul>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
